### PR TITLE
Bug/#113 sidekiq variant bug fixes

### PR DESCRIPTION
**Pull request by Rabid-Dan**
Addresses #113

- Removes redis from being added as an explicit dependancy in the sidekiq template as sidekiq itself already brings it in as a dependancy.
- Changes route insertion in the sidekiq template to use route syntax as the presence of multiple insert_into_file calls was causing the route addition to fail during app generation (meaning it just wouldnt be added to the routes.rb file).
- Removes `Sidekiq::Web.app_url` from the sidekiq initialiser template as it wasnt needed and a route for the web panel was already being added to the routes file.

The only remaining issue from #113 was to not add sidekiq as the default queue adapter. I did a bit of testing and couldnt replicate the issue where redis was required to run tests but if someone is able to explain to me either how I could replicate it or confirm that it's still something that needs changing then I'm happy to change that line addition to a comment or remove it completely.

### CODE REVIEW

File: variants/sidekiq/config/initializers/sidekiq.rb.tt [hunk 1]
```diff
@@ -3,7 +3,6 @@ require "sidekiq/web"
 sidekiq_username = ENV.fetch("SIDEKIQ_WEB_USERNAME")
 sidekiq_password = ENV.fetch("SIDEKIQ_WEB_PASSWORD")
 
-Sidekiq::Web.app_url = "/"
```
[the remark this answers is not in the document]
Comment (Rabid-Dan) [reply variants/sidekiq/config/initializers/sidekiq.rb.tt:6]: 👍  I'd be cool with that. I tried to figure out what it was supposed to do but I couldn't find any docs around it. Leaving it in as a commented line feels like a good compromise.

File: variants/sidekiq/template.rb [hunk 2]
```diff
@@ -41,9 +40,7 @@
     image: redis
 ", after: /^services:$"/
 
-insert_into_file "config/routes.rb", before: "root to: \"home#index\"" do
-  <<~ROUTE
-
+route <<~ROUTE
```
[the remark this answers is not in the document]
Comment (Rabid-Dan) [reply variants/sidekiq/template.rb:43]: I've done a trial run of generating an app with it and it does add it to the top but everything still seems to function fine when its running. That could change if more routes are added I guess but for now it all seems functional

File: variants/sidekiq/config/initializers/sidekiq.rb.tt (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshmcarthur) [reply variants/sidekiq/config/initializers/sidekiq.rb.tt:6]: Looks like it's used in the Sidekiq nav: https://github.com/mperham/sidekiq/blob/master/web/views/_nav.erb#L12

File: variants/sidekiq/template.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshmcarthur) [reply variants/sidekiq/template.rb:43]: :+1:

**Pull request merged**